### ISSUE TITLE
pkg_add can't find things when run from config.vm.provision

### ISSUE THREAD

**Issue by mcandre** [tmatilai/packer-openbsd#30]
`sudo pkg_add` works in a manual `vagrant ssh` session, but when I try to automate installing the same packages by specifying a `config.vm.provision` in the Vagrantfile, for some reason `pkg_add` fails.

Perhaps there is some shell configuration that is not being fully loaded in non-interactive mode?

**Comment by mcandre** [comment]
Ah, `PKG_PATH` is being configured by /home/vagrant/.profile, which appears to not be loaded by non-interactive sessions, which includes `config.vm.provision`.

As a workaround, I am explicitly setting `PKG_PATH` when I run pkg_add commands:

Vagrantfile:

```ruby
Vagrant.configure("2") do |config|
  config.vm.box = "ryanmaclean/openbsd-6.0"
  config.vm.provision "shell", path: "bootstrap.sh"
end
```

bootstrap.sh:

```sh
#!/bin/sh
sudo PKG_PATH=http://ftp.openbsd.org/pub/OpenBSD/6.0/packages/amd64/ pkg_add rust cargo
```

But it would be nice if `PKG_PATH` were available by default for all sessions, including non-interactive sessions.

**Comment by dentarg** [comment]
> Perhaps there is some shell configuration that is not being fully loaded in non-interactive mode?

Yes, from https://www.vagrantup.com/docs/provisioning/shell.html:

> For POSIX-like machines, the shell provisioner executes scripts with SSH.

`~/.profile` isn't sourced for non-interactive / non-login login shells, for [various](https://unix.stackexchange.com/questions/257571/why-does-bashrc-check-whether-the-current-shell-is-interactive) [reasons](https://unix.stackexchange.com/questions/38175/difference-between-login-shell-and-non-login-shell). `PKG_PATH` need to be set with `env` argument to the shell provisioner.

**Comment by dentarg** [comment]
Another option is to use [`installurl(5)`](https://man.openbsd.org/installurl), not that well documented that it also works for [`pkg_add`](https://man.openbsd.org/OpenBSD-6.1/pkg_add), but it does.
